src/app/utils/app-utils.ts

```diff
@@ -178,4 +179,32 @@ export class AppUtils {
     return globalError;
   }
 
+  // Util method to return Navitems
+  // based on user's role
+  public static getNavItemsBasedOnRole(roleBasedNav: UserRoleNav, userRoles: string[]): NavItem[] {
+    let roleNavItems: NavItem [] = new Array<NavItem>();
+    userRoles.forEach(role => {
+      if (roleBasedNav.hasOwnProperty(role)) {
+        roleNavItems = [...roleNavItems, roleBasedNav[role]];
+      }
+    });
+    return roleNavItems.sort((a, b) => (a.orderId > b.orderId) ? 1 : -1)
```

Comment:
does the sort need support a.orderId === b.orderId?

Reply:
@rayliangatos I could not get your question?

Reply:
Fixed as discussed with you @rayliangatos